tests/ui/lint/non-local-defs/convoluted-locals-132427.rs

```diff
@@ -0,0 +1,54 @@
+// Regression tests for https://github.com/rust-lang/rust/issues/132427
+
+//@ check-pass
+
+// original
+mod auth {
+    const _: () = {
+        pub enum ArbitraryContext {}
+
+        const _: () = {
+            impl ArbitraryContext {}
+        };
```

Comment:
Remark (for myself): does this peeling logic work for `macro_generated!` inner impls? I think it should, because it operates based on `DefId`s AFAICT and not spans.

Reply:
Yes, it does work for macro-generated code, for the reason you explained.
Fro example #132427 was from a proc-macro.